### PR TITLE
update for batch upserts in collections

### PR DESCRIPTION
**Pull request by jairad26**

### CODE REVIEW

File: src/assembly/collections.ts [hunk 1]
```diff
@@ -22,18 +22,22 @@ abstract class CollectionResult {
...
+    keys: string[],
   ) {
     super(collection, status, error);
     this.operation = operation;
-    this.key = key;
+    if (keys == null) {
+      this.keys = [];
+    } else {
+      this.keys = keys;
+    }
```
Comment (mattjohnsonpint) [inline src/assembly/collections.ts:40]:
```suggestion
    this.keys = keys;    
```

`keys` can't be null here.

File: examples/emojisearch/assembly/index.ts [hunk 2]
```diff
@@ -62,30 +62,29 @@ function generateListForEmoji(text: string): string[] {
...
-  for (let i: i32 = 0; i < descriptions.length; i++) {
-    response = insertEmoji(descriptions[i]);
-    if (response !== "success") {
-      return response;
+  // Upsert emojis in batches of 50
+  for (let i: i32 = 0; i < emojiList.length; i += upsertBatchSize) {
+    const end: i32 = min(i + upsertBatchSize, emojiList.length);
+    const batch: string[] = emojiList.slice(i, end);
+    const response = collections.upsertBatch(emojis, null, batch);
+    if (response.isSuccessful === false) {
```
Comment (mattjohnsonpint) [inline examples/emojisearch/assembly/index.ts:82]: nit:
```suggestion
    if (!response.isSuccessful) {
```

File: src/assembly/collections.ts [hunk 3]
```diff
@@ -159,18 +207,25 @@ export function upsert(
...
+  const result = hostUpsertToCollection(collection, keys, texts);
   if (utils.resultIsInvalid(result)) {
     console.error("Error upserting to Text index.");
     return new CollectionMutationResult(
       collection,
       CollectionStatus.Error,
       "Error upserting to Text index.",
       "upsert",
-      "",
+      [],
```
Comment (mattjohnsonpint) [inline src/assembly/collections.ts:228]: All these empty `[]` feel strange.  Why not just make it optional on the constructor instead?

Comment (jairad26) [reply src/assembly/collections.ts:228]: i dont want to have to run into some unexpected null issues we can do this later

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**